model/riscv_sys_regs.sail

```diff
@@ -736,3 +752,71 @@ function legalize_satp32(a : Architecture, o : bits(32), v : bits(32)) -> bits(3
 
 /* disabled trigger/debug module */
 register tselect : xlenbits
+
+
+/*
+ * The seed CSR (entropy source)
+ * ------------------------------------------------------------
+ */
+
+/* Valid return states for reading the seed CSR. */
+enum seed_opst = {
+  BIST, // Built-in-self-test. No randomness sampled.
+  ES16, // Entropy-sample-16. Valid 16-bits of randomness sampled.
+  WAIT, // Device still gathering entropy.
+  DEAD  // Fatal device compromise. No randomness sampled.
+}
+
+/* Mapping of status codes and their actual encodings. */
+mapping opst_code : seed_opst <-> bits(2) = {
+  BIST <-> 0b00,
+  WAIT <-> 0b01,
+  ES16 <-> 0b10,
+  DEAD <-> 0b11
+}
+
+/*
+ * Entropy Source - Platform access to random bits.
+ * WARNING: This function currently lacks a proper side-effect annotation.
+ *          If you are using theorem prover tool flows, you
+ *          may need to modify or stub out this function for now.
+ * NOTE: This would be better placed in riscv_platform.sail, but that file
```

Comment:
I hope this is an acceptable compromise? I moved the Zkr code to be among the other CSR instructions. By the time I'd fixed the use of `ext_*` functions, there seemed no point it living in a separate file. Happy changing this if there's a better way.

Reply:
This is fine, though can the platform function not just return `bits(16)` and thus you just have `val get_16_random_bits = { ... }` and no need for a wrapper? Then the naming matches other functions in this file, some of which do still use platform functions but don't include a plat_ prefix in their Sail names.

Reply:
There _was_ once a reason for having a separate wrapper function but I think it went away as I learned to Sail. So yes, wrapper removed and `get_16_random_bits` now returns `bits(16)`.